### ISSUE TITLE
RunAsync is taking longer then expected time (5s) to cancel.

### ISSUE THREAD

**Issue by sondreb** [Azure-Samples/service-fabric-dotnet-web-reference-app#25]
On line **573** in **InventoryService.cs**, there is a delay that does not take the cancellationToken. I think the default backupFrequencyInSeconds is 10 seconds, and that is longer than cancellation is suppose to take (5 seconds). So on upgrades, there appears warnings for every service that uses the sample for taking periodic backups.

This existing code:
`await Task.Delay(TimeSpan.FromSeconds(this.backupManager.backupFrequencyInSeconds));`

Should be modified to:

`await Task.Delay(TimeSpan.FromSeconds(this.backupManager.backupFrequencyInSeconds), cancellationToken);`

I have tested and verified that this removes the warning during upgrade of clusters.

**Comment by masnider** [comment]
Yes, that looks correct and good catch on the bug (I don't know why I passed in the CT and then ignored it) Either you can submit a PR or you can wait for me to "officially" fix it.
